src/server/game/Scripting/ScriptMgr.h

```diff
@@ -787,6 +787,9 @@ class AchievementCriteriaScript : public ScriptObject
 
         bool IsDatabaseBound() const { return true; }
 
+// TODO: manually silencing this warning, but it should properly fixed
+#pragma GCC diagnostic ignored "-Woverloaded-virtual"
```

Comment:
https://github.com/azerothcore/azerothcore-wotlk/pull/3473/files

Reply:
@Winfidonarleyan thank you